components/tidb_query_datatype/src/codec/batch/lazy_column_vec.rs

```diff
@@ -185,16 +187,58 @@ impl LazyBatchColumnVec {
 // Do not implement Deref, since we want to forbid some misleading function calls like
 // `LazyBatchColumnVec.len()`.
 
-impl std::ops::Index<usize> for LazyBatchColumnVec {
+impl Index<usize> for LazyBatchColumnVec {
     type Output = LazyBatchColumn;
 
     fn index(&self, index: usize) -> &LazyBatchColumn {
         &self.columns[index]
     }
 }
 
-impl std::ops::IndexMut<usize> for LazyBatchColumnVec {
+impl IndexMut<usize> for LazyBatchColumnVec {
     fn index_mut(&mut self, index: usize) -> &mut LazyBatchColumn {
         &mut self.columns[index]
     }
 }
+
+impl Index<Range<usize>> for LazyBatchColumnVec {
```

Comment:
Perhaps `Index<Range<usize>` is enough since we don't need method like [`SliceIndex::get_unchecked()`](https://doc.rust-lang.org/std/slice/trait.SliceIndex.html#tymethod.get_unchecked)